api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/ContrastAccountServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+        LOGGER.info("Create secret response: {}", createResponse);
+        String accountId = UUID.randomUUID().toString();
+        createAccountInDb(accountData.getContrastOrganizationId(), organizationName, CONTRAST, accountData.getCreatedBy());
+        updateConfigProperty(CONTRAST_ENABLED, TRUE, JOB_SCHEDULER);
+        sendSQSMessage(CONTRAST, accountData.getContrastOrganizationId());
+        validationResponse = new AccountValidationResponse();
+        validationResponse.setValidationStatus(SUCCESS);
+        validationResponse.setAccountId(accountId);
+        validationResponse.setAccountName(organizationName);
+        validationResponse.setType(CONTRAST);
+        validationResponse.setMessage("Account added successfully. ID: " + accountId);
+        return validationResponse;
+    }
+
+    private AccountValidationResponse validateRequestData(CreateAccountRequest accountData) {
+        AccountValidationResponse response = new AccountValidationResponse();
+        List<String> missingParams = new ArrayList<>();
+        if (StringUtils.isEmpty(accountData.getContrastOrganizationId())) {
+            missingParams.add("organizationId");
+        }
+        if (StringUtils.isEmpty(accountData.getContrastServiceKey())) {
+            missingParams.add("serviceKey");
+        }
+        if (StringUtils.isEmpty(accountData.getContrastApiKey())) {
+            missingParams.add("apiKey");
+        }
+        if (StringUtils.isEmpty(accountData.getContrastUserId())) {
+            missingParams.add("userId");
+        }
+        if (!missingParams.isEmpty()) {
+            String errorMessage = MISSING_MANDATORY_PARAMETER + String.join(", ", missingParams);
+            response.setMessage(errorMessage);
+            response.setValidationStatus(FAILURE);
+        } else {
+            response.setValidationStatus(SUCCESS);
+        }
+        return response;
+    }
+
+    @Override
+    public AccountValidationResponse deleteAccount(String organizationId) {
+        LOGGER.info("Deleting contrast with organizationId: {}", organizationId);
+        AccountValidationResponse response = new AccountValidationResponse();
+        DeleteSecretResult deleteResponse = deleteSecret(organizationId, CONTRAST);
+        LOGGER.info("Delete secret response: {} ", deleteResponse);
+        deleteAccountFromDB(organizationId);
+        updateConfigProperty(CONTRAST_ENABLED, FALSE, JOB_SCHEDULER);
+        response.setType(CONTRAST);
+        response.setAccountId(organizationId);
+        response.setValidationStatus(SUCCESS);
+        response.setMessage("Account deleted successfully");
+        return response;
+    }
+
+    private String getContrastSecretString(CreateAccountRequest accountRequest) {
+        String authKey = Base64.getEncoder().encodeToString((accountRequest.getContrastUserId() + ":"
+                + accountRequest.getContrastServiceKey()).getBytes());
```

Comment:
I see this code twice here. Make it a small method `getAuthorizationHeader`.

Reply:
changed